kasa/tests/conftest.py

```diff
@@ -172,135 +242,136 @@ def _make_unsupported(device_family, encrypt_type):
+        return paramtuple.name + (
+            "" if paramtuple.protocol == "IOT" else "-" + paramtuple.protocol
         )
     except:  # TODO: HACK as idgenerator is now used by default  # noqa: E722
         return None
 
 
-def filter_model(desc, model_filter, protocol_filter=None):
-    if protocol_filter is None:
-        protocol_filter = {"IOT", "SMART"}
-    filtered = list()
-    for file, protocol in SUPPORTED_DEVICES:
-        if protocol in protocol_filter:
-            file_model_region = basename(file).split("_")[0]
-            file_model = file_model_region.split("(")[0]
-            for model in model_filter:
-                if model == file_model:
-                    filtered.append((file, protocol))
-
-    filtered_basenames = [basename(f) + "-" + p for f, p in filtered]
-    print(f"# {desc}")
-    for file in filtered_basenames:
-        print(f"\t{file}")
-    return filtered
-
-
-def parametrize(desc, devices, protocol_filter=None, ids=None):
+def parametrize(
+    desc,
+    *,
+    model_filter=None,
+    protocol_filter=None,
+    component_filter=None,
+    data_root_filter=None,
+    ids=None,
+):
     if ids is None:
         ids = idgenerator
     return pytest.mark.parametrize(
-        "dev", filter_model(desc, devices, protocol_filter), indirect=True, ids=ids
+        "dev",
+        filter_fixtures(
+            desc,
+            model_filter=model_filter,
+            protocol_filter=protocol_filter,
+            component_filter=component_filter,
+            data_root_filter=data_root_filter,
+        ),
+        indirect=True,
+        ids=ids,
     )
 
 
-has_emeter = parametrize("has emeter", WITH_EMETER, protocol_filter={"SMART", "IOT"})
+has_emeter = parametrize(
+    "has emeter", model_filter=WITH_EMETER, protocol_filter={"SMART", "IOT"}
+)
 no_emeter = parametrize(
-    "no emeter", ALL_DEVICES - WITH_EMETER, protocol_filter={"SMART", "IOT"}
+    "no emeter",
+    model_filter=ALL_DEVICES - WITH_EMETER,
+    protocol_filter={"SMART", "IOT"},
+)
+has_emeter_iot = parametrize(
+    "has emeter iot", model_filter=WITH_EMETER_IOT, protocol_filter={"IOT"}
 )
-has_emeter_iot = parametrize("has emeter iot", WITH_EMETER_IOT, protocol_filter={"IOT"})
 no_emeter_iot = parametrize(
-    "no emeter iot", ALL_DEVICES_IOT - WITH_EMETER_IOT, protocol_filter={"IOT"}
+    "no emeter iot",
+    model_filter=ALL_DEVICES_IOT - WITH_EMETER_IOT,
+    protocol_filter={"IOT"},
 )
 
-bulb = parametrize("bulbs", BULBS, protocol_filter={"SMART", "IOT"})
-plug = parametrize("plugs", PLUGS, protocol_filter={"IOT"})
-strip = parametrize("strips", STRIPS, protocol_filter={"SMART", "IOT"})
-dimmer = parametrize("dimmers", DIMMERS, protocol_filter={"IOT"})
-lightstrip = parametrize("lightstrips", LIGHT_STRIPS, protocol_filter={"IOT"})
+bulb = parametrize("bulbs", model_filter=BULBS, protocol_filter={"SMART", "IOT"})
+plug = parametrize("plugs", model_filter=PLUGS, protocol_filter={"IOT"})
```

Comment:
I split this into `device_fixtures` and `discovery_fixtures`. We could split `device_fixtures` further but I'm not sure whether `_iot` and `_smart` is the right way to do it because when have iot supporting features we'll want fixtures that have both.